### PR TITLE
Dropdown for Testing facility Name

### PR DESCRIPTION
**Pull request by bodhisha**

### CODE REVIEW

**Comment by tomahawk-pilot** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 20
- Added 16
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652&bid=19084232)

File: src/Components/Patient/SampleTest.tsx [hunk 1]
```diff
@@ -105,6 +143,12 @@ export const SampleTest = (props: any) => {
             invalidForm = true;
           }
           break;
+        case "testing_facility":
+          if (!state.form[field]) {
+            errors[field] = "Please Choose a testing facility";
```
Comment (tomahawk-pilot) [inline src/Components/Patient/SampleTest.tsx:148]: Codacy found an issue: [Generic Object Injection Sink](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/SampleTest.tsx [hunk 2]
```diff
@@ -25,6 +30,7 @@ const initForm: SampleTestModel = {
   diagnosis: "",
   diff_diagnosis: "",
   doctor_name: "",
+  testing_facility: "",
```
Comment (tomahawk-pilot) [inline src/Components/Patient/SampleTest.tsx:33]: Codacy found an issue: [Identifier 'testing_facility' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/SampleTest.tsx [hunk 3]
```diff
@@ -66,10 +75,39 @@ export const SampleTest = (props: any) => {
...
+      const FacilityNameList = await dispatchAction(
+        getFacilities({ facility_type })
+      );
+      if (!status.aborted && FacilityNameList.data.results) {
+        setFacilityName([...FacilityNameList.data.results]);
+        dispatch({
+          type: "set_form",
+          form: {
+            ...state.form,
+            testing_facility: FacilityNameList.data.results[0]?.id,
```
Comment (tomahawk-pilot) [inline src/Components/Patient/SampleTest.tsx:97]: Codacy found an issue: [Identifier 'testing_facility' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/SampleTest.tsx [hunk 4]
```diff
@@ -131,21 +175,33 @@ export const SampleTest = (props: any) => {
...
         is_atypical_presentation: state.form.is_atypical_presentation,
-        atypical_presentation: state.form.is_atypical_presentation ? state.form.atypical_presentation : undefined,
+        atypical_presentation: state.form.is_atypical_presentation
+          ? state.form.atypical_presentation
+          : undefined,
         diagnosis: state.form.diagnosis ? state.form.diagnosis : undefined,
-        diff_diagnosis: state.form.diff_diagnosis ? state.form.diff_diagnosis : undefined,
-        doctor_name: state.form.doctor_name ? state.form.doctor_name : undefined,
-        etiology_identified: state.form.etiology_identified ? state.form.etiology_identified : undefined,
+        diff_diagnosis: state.form.diff_diagnosis
```
Comment (tomahawk-pilot) [inline src/Components/Patient/SampleTest.tsx:182]: Codacy found an issue: [Identifier 'diff_diagnosis' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/SampleTest.tsx [hunk 5]
```diff
@@ -66,10 +75,39 @@ export const SampleTest = (props: any) => {
...
+  );
 
   const headerText = "Request Sample";
   const buttonText = "Confirm your request to send sample for testing";
 
+  const fetchFacilityName = useCallback(
+    async (status: statusType) => {
+      const facility_type = 950;
+      const FacilityNameList = await dispatchAction(
+        getFacilities({ facility_type })
```
Comment (tomahawk-pilot) [inline src/Components/Patient/SampleTest.tsx:89]: Codacy found an issue: [Identifier 'facility_type' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/SampleTest.tsx [hunk 6]
```diff
@@ -131,21 +175,33 @@ export const SampleTest = (props: any) => {
...
+          : undefined,
         diagnosis: state.form.diagnosis ? state.form.diagnosis : undefined,
-        diff_diagnosis: state.form.diff_diagnosis ? state.form.diff_diagnosis : undefined,
-        doctor_name: state.form.doctor_name ? state.form.doctor_name : undefined,
-        etiology_identified: state.form.etiology_identified ? state.form.etiology_identified : undefined,
+        diff_diagnosis: state.form.diff_diagnosis
+          ? state.form.diff_diagnosis
+          : undefined,
+        testing_facility: state.form.testing_facility,
+        doctor_name: state.form.doctor_name
```
Comment (tomahawk-pilot) [inline src/Components/Patient/SampleTest.tsx:186]: Codacy found an issue: [Identifier 'doctor_name' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/models.tsx [hunk 7]
```diff
@@ -1,188 +1,205 @@
...
-    email?: string;
+  email?: string;
 }
 
 export interface PatientModel {
-    id?: number;
+  id?: number;
+  name?: string;
+  age?: number;
+  allow_transfer?: boolean;
```
Comment (tomahawk-pilot) [inline src/Components/Patient/models.tsx:20]: Codacy found an issue: [Identifier 'allow_transfer' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/SampleTest.tsx [hunk 8]
```diff
@@ -131,21 +175,33 @@ export const SampleTest = (props: any) => {
...
+          ? state.form.doctor_name
+          : undefined,
+        etiology_identified: state.form.etiology_identified
```
Comment (tomahawk-pilot) [inline src/Components/Patient/SampleTest.tsx:189]: Codacy found an issue: [Identifier 'etiology_identified' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/SampleTest.tsx [hunk 9]
```diff
@@ -82,19 +120,19 @@ export const SampleTest = (props: any) => {
...
         case "icmr_label":
-          if ( !state.form[field]) {
+          if (!state.form[field]) {
             errors[field] = "Please specify the label";
             invalidForm = true;
           }
           break;
         case "sample_type_other":
-          if (state.form.sample_type === 'OTHER TYPE' && !state.form[field]) {
+          if (state.form.sample_type === "OTHER TYPE" && !state.form[field]) {
```
Comment (tomahawk-pilot) [inline src/Components/Patient/SampleTest.tsx:135]: Codacy found an issue: [Generic Object Injection Sink](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/SampleTest.tsx [hunk 10]
```diff
@@ -131,21 +175,33 @@ export const SampleTest = (props: any) => {
...
+        doctor_name: state.form.doctor_name
+          ? state.form.doctor_name
+          : undefined,
+        etiology_identified: state.form.etiology_identified
+          ? state.form.etiology_identified
+          : undefined,
         sample_type: state.form.sample_type,
         icmr_category: state.form.icmr_category,
-        sample_type_other: state.form.sample_type === 'OTHER TYPE' ? state.form.sample_type_other : undefined,
+        sample_type_other:
```
Comment (tomahawk-pilot) [inline src/Components/Patient/SampleTest.tsx:194]: Codacy found an issue: [Identifier 'sample_type_other' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/SampleTest.tsx [hunk 11]
```diff
@@ -131,21 +175,33 @@ export const SampleTest = (props: any) => {
         has_sari: state.form.has_sari,
         is_unusual_course: state.form.is_unusual_course,
         is_atypical_presentation: state.form.is_atypical_presentation,
-        atypical_presentation: state.form.is_atypical_presentation ? state.form.atypical_presentation : undefined,
+        atypical_presentation: state.form.is_atypical_presentation
+          ? state.form.atypical_presentation
+          : undefined,
```
Comment (tomahawk-pilot) [inline src/Components/Patient/SampleTest.tsx:180]: Codacy found an issue: [Unexpected use of undefined.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/SampleTest.tsx [hunk 12]
```diff
@@ -131,21 +175,33 @@ export const SampleTest = (props: any) => {
         has_sari: state.form.has_sari,
         is_unusual_course: state.form.is_unusual_course,
         is_atypical_presentation: state.form.is_atypical_presentation,
-        atypical_presentation: state.form.is_atypical_presentation ? state.form.atypical_presentation : undefined,
+        atypical_presentation: state.form.is_atypical_presentation
```
Comment (tomahawk-pilot) [inline src/Components/Patient/SampleTest.tsx:178]: Codacy found an issue: [Identifier 'atypical_presentation' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/models.tsx [hunk 13]
```diff
@@ -1,188 +1,205 @@
...
+  medical_history?: Array<{ disease: string | number; details: string }>;
+  facility_object?: {
+    id: number;
+    name: string;
+    facility_type?: { id: number; name: string };
+  };
+  contact_with_carrier?: boolean;
+  medical_history_details?: string;
+  is_active?: boolean;
+  local_body_object?: { id: number; name: string };
```
Comment (tomahawk-pilot) [inline src/Components/Patient/models.tsx:34]: Codacy found an issue: [Identifier 'local_body_object' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/models.tsx [hunk 14]
```diff
@@ -1,188 +1,205 @@
...
+  allow_transfer?: boolean;
+  discharge?: boolean;
+  gender?: number;
+  phone_number?: string;
+  allergies?: string;
+  medical_history?: Array<{ disease: string | number; details: string }>;
+  facility_object?: {
+    id: number;
+    name: string;
+    facility_type?: { id: number; name: string };
```
Comment (tomahawk-pilot) [inline src/Components/Patient/models.tsx:29]: Codacy found an issue: [Identifier 'facility_type' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/SampleTest.tsx [hunk 15]
```diff
@@ -131,21 +175,33 @@ export const SampleTest = (props: any) => {
...
+          ? state.form.atypical_presentation
+          : undefined,
         diagnosis: state.form.diagnosis ? state.form.diagnosis : undefined,
-        diff_diagnosis: state.form.diff_diagnosis ? state.form.diff_diagnosis : undefined,
-        doctor_name: state.form.doctor_name ? state.form.doctor_name : undefined,
-        etiology_identified: state.form.etiology_identified ? state.form.etiology_identified : undefined,
+        diff_diagnosis: state.form.diff_diagnosis
+          ? state.form.diff_diagnosis
+          : undefined,
+        testing_facility: state.form.testing_facility,
```
Comment (tomahawk-pilot) [inline src/Components/Patient/SampleTest.tsx:185]: Codacy found an issue: [Identifier 'testing_facility' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/SampleTest.tsx [hunk 16]
```diff
@@ -66,10 +75,39 @@ export const SampleTest = (props: any) => {
...
+  const [facilityName, setFacilityName] = useState<Array<FacilityNameModel>>(
+    []
+  );
 
   const headerText = "Request Sample";
   const buttonText = "Confirm your request to send sample for testing";
 
+  const fetchFacilityName = useCallback(
+    async (status: statusType) => {
+      const facility_type = 950;
```
Comment (tomahawk-pilot) [inline src/Components/Patient/SampleTest.tsx:87]: Codacy found an issue: [Identifier 'facility_type' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

**Comment by dauntlessnomad** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 20
- Added 16
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652&bid=19084232)

File: src/Components/Patient/SampleTest.tsx (hunk 5 above)
Comment (dauntlessnomad) [inline src/Components/Patient/SampleTest.tsx:89]: Codacy found an issue: [Identifier 'facility_type' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/SampleTest.tsx (hunk 9 above)
Comment (dauntlessnomad) [inline src/Components/Patient/SampleTest.tsx:135]: Codacy found an issue: [Generic Object Injection Sink](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/SampleTest.tsx (hunk 8 above)
Comment (dauntlessnomad) [inline src/Components/Patient/SampleTest.tsx:189]: Codacy found an issue: [Identifier 'etiology_identified' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/SampleTest.tsx (hunk 15 above)
Comment (dauntlessnomad) [inline src/Components/Patient/SampleTest.tsx:185]: Codacy found an issue: [Identifier 'testing_facility' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/SampleTest.tsx (hunk 3 above)
Comment (dauntlessnomad) [inline src/Components/Patient/SampleTest.tsx:97]: Codacy found an issue: [Identifier 'testing_facility' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/SampleTest.tsx (hunk 2 above)
Comment (dauntlessnomad) [inline src/Components/Patient/SampleTest.tsx:33]: Codacy found an issue: [Identifier 'testing_facility' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/SampleTest.tsx (hunk 1 above)
Comment (dauntlessnomad) [inline src/Components/Patient/SampleTest.tsx:148]: Codacy found an issue: [Generic Object Injection Sink](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/SampleTest.tsx (hunk 11 above)
Comment (dauntlessnomad) [inline src/Components/Patient/SampleTest.tsx:180]: Codacy found an issue: [Unexpected use of undefined.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/SampleTest.tsx (hunk 6 above)
Comment (dauntlessnomad) [inline src/Components/Patient/SampleTest.tsx:186]: Codacy found an issue: [Identifier 'doctor_name' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/models.tsx (hunk 7 above)
Comment (dauntlessnomad) [inline src/Components/Patient/models.tsx:20]: Codacy found an issue: [Identifier 'allow_transfer' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/SampleTest.tsx (hunk 10 above)
Comment (dauntlessnomad) [inline src/Components/Patient/SampleTest.tsx:194]: Codacy found an issue: [Identifier 'sample_type_other' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/SampleTest.tsx (hunk 16 above)
Comment (dauntlessnomad) [inline src/Components/Patient/SampleTest.tsx:87]: Codacy found an issue: [Identifier 'facility_type' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/models.tsx (hunk 14 above)
Comment (dauntlessnomad) [inline src/Components/Patient/models.tsx:29]: Codacy found an issue: [Identifier 'facility_type' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/SampleTest.tsx (hunk 4 above)
Comment (dauntlessnomad) [inline src/Components/Patient/SampleTest.tsx:182]: Codacy found an issue: [Identifier 'diff_diagnosis' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/SampleTest.tsx (hunk 12 above)
Comment (dauntlessnomad) [inline src/Components/Patient/SampleTest.tsx:178]: Codacy found an issue: [Identifier 'atypical_presentation' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

File: src/Components/Patient/models.tsx (hunk 13 above)
Comment (dauntlessnomad) [inline src/Components/Patient/models.tsx:34]: Codacy found an issue: [Identifier 'local_body_object' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5828652)

**Pull request merged**